pxr/imaging/plugin/hdRpr/diskLight.cpp

```diff
@@ -0,0 +1,58 @@
+#include "diskLight.h"
+#include <cmath>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+const TfTokenVector k_requiredGeometryParam = {
+    HdLightTokens->radius,
+};
+
+bool HdRprDiskLight::IsDirtyGeomParam(std::map<TfToken, float>& params) {
+    auto radiusParamIter = params.find(HdLightTokens->radius);
+    if (radiusParamIter == params.end()) {
+        return false;
+    }
+
+    float radius = radiusParamIter->second;
+
+    bool isDirty = radius != m_radius;
+
+    m_radius = radius;
+
+    return isDirty;
+}
+
+const TfTokenVector& HdRprDiskLight::FetchLightGeometryParamNames() const {
+    return k_requiredGeometryParam;
+}
+
+RprApiObjectPtr HdRprDiskLight::CreateLightMesh(std::map<TfToken, float>& params) {
+    HdRprApiSharedPtr rprApi = m_rprApiWeakPtr.lock();
+    if (!rprApi) {
+        TF_CODING_ERROR("RprApi is expired");
+        return nullptr;
+    }
+
+    return rprApi->CreateDiskLightMesh(params[HdLightTokens->radius]);
+}
+
+GfVec3f HdRprDiskLight::NormalizeLightColor(const GfMatrix4d& transform, std::map<TfToken, float>& params, const GfVec3f& inColor) {
+    float radius = params[HdLightTokens->radius];
+    if (radius <= 0.0f) {
+        radius = 1.0f;
```

Comment:
Should this be radius = -radius?

Reply:
Rather I'm saying any -radius light should be automatically treated as the negative of the negative radius

Reply:
Maybe. This logic is a copypaste from existing logic in sphere light. If we want to take the absolute value of the radius here then for sphere light we should the same